File: frontend/src/features/posts/services/postService.ts
import api from '@/lib/axios';

export const postService = {
    getPosts: async (page = 1) => {
        const response = await api.get(`/posts?page=${page}`);
        return response.data;
    },
    createPost: async (data: any) => {
        const response = await api.post('/posts', data);
        return response.data;
    },
    likePost: async (postId: string) => {
        const response = await api.post(`/posts/${postId}/like`);
        return response.data;
    }
};

File: backend/post-service/src/main/java/com/cine/social/post/service/PostService.java
package com.cine.social.post.service;



import com.cine.social.common.dto.response.PageResponse;
import com.cine.social.post.dto.request.PostCreationRequest;
import com.cine.social.post.dto.request.PostUpdateRequest;
import com.cine.social.post.dto.response.PostResponse;

import java.util.List;

public interface PostService {

    PostResponse createPost(PostCreationRequest request);
    void updatePost(String id, PostUpdateRequest request);

    void deletePost(String postId);

    void votePost(String postId, int value);

    PageResponse<List<PostResponse>> getMyPosts(int page, int size);

    PageResponse<List<PostResponse>> getPosts(int page, int size);

    void retryPost(String postId);

    PageResponse<List<PostResponse>> getMyFeed(int page, int size);

    PageResponse<List<PostResponse>> getReels(int page, int size);
}
